### DISCUSSION TITLE
Delete Data Source from PBI Model?

### DISCUSSION THREAD

**Discussion by Ahad-boop** [TabularEditor/TabularEditor3#1159] [Data Modeling]
Hi! 

I have a problem that one of my data sources for the model got deleted form the back end and replaced with a new one. now tabular editor still shows the old one and now i cannot refresh the model as it asks for credentials for the old one. I am trying to remove it but there is no way i see so far. there are no connections in the data sources folder that i can delete. none of the tables are pointing to the old endpoint. i double checked all the m code of all the tables. so how do i fix this?

**Comment by otykier** [comment]
Hi @Ahad-boop 

Most likely you still have one or more partitions that explicitly reference the data source, preventing it from being deleted. To detect which partitions still reference it, you can do the following:

1. Open a new C# script (**File > New > C# script**)
2. Paste in the following code: `Selected.DataSource.UsedByPartitions.Output();`
3. Select the data source in the TOM Explorer, then hit F5 to run the script.
